### DISCUSSION TITLE
Filter out specific result domains, for example, block all pinterest?

### DISCUSSION THREAD

**Discussion by wbxqm** [searx/searx#3107] [General]
Hi all, 

Is it possible to filter results from search engines based on the domain that the result is from? For example, would it be possible to not see any Pinterest-sourced results from any of the search engines?

**Comment by neeerp** [comment]
Was about to ask the same question as some preliminary searching didn't seem to answer this for me. Here's some context on why I want this, if anyone cares:

I have a browser extension for DDG where I can click to hide a specific domain (e.g. github and SO scraper sites that clutter results). The problem is that this filtering is client sided, so my results page just ends up being shorter. In the extreme case, the entire first page of results is filtered. I was thinking I could just run my own searx instance and do this same filtering on the server side and render a full page of filtered results.

**Reply by 0xhtml** [reply]
There is quite some discussion already in the issue #2001. Take a look there.

To my knowledge searx is subject to the same issue as described with the browser extension and DDG. searx is a meta-search engine and therefor is making requests to other search engines that serve the actual results. If searx gets for example 10 results from Google and all 10 results from google are filtered out by the defined block list, searx wouldn't have any results to display on the first page (assuming all other engines return empty handed as well). In order to fill up the first page searx would need to send another request to Google to get more results which then would be filtered again. This would have to go on until let's say 10 results that weren't filtered out have accumulated. All of this could greatly increase the loading time of the search.

Another problem is faced if the user then wants to view the second page. searx would have to remember on which page of Google's result pages to continue, because this wouldn't be the second page as searx already used multiple pages to fill up the first page of searx's result pages.

If anyone is interested in implementing a plugin for searx to filter results there are some instructions here #1737. However a plugin is AFAIK not able to do the extra requests previously described.

**Reply by dalf** [reply]
This is implemented in SearXNG : https://github.com/searxng/searxng/blob/7966fd3bbdee448d5f4b532231d69310b4f91563/searx/settings.yml#L178-L186

**Reply by kvch** [reply]
> This is implemented in SearXNG : https://github.com/searxng/searxng/blob/7966fd3bbdee448d5f4b532231d69310b4f91563/searx/settings.yml#L178-L186

For the record, it is implemented is searx as well: https://github.com/searx/searx/pull/2724
